### PR TITLE
Add JavaScript for research survey

### PR DESCRIPTION
**Pull request by jacksonj04**
Add JavaScript for tracking users and performing logic, and page elements to send users to the survey.

The JavaScript is a compressed version of the full code found at https://gist.github.com/jacksonj04/a84f4b0d38d02b6cee6d/3439b2c8601ec08408de0eb0e801bd107d81df0e.

### CODE REVIEW

File: templates/web/base/report/display.html [hunk 1]
```diff
@@ -20,6 +20,8 @@
     <p class='form-success'>[% loc('You have successfully signed in; please check and confirm your details are accurate:') %]</p>
 [% END %]
 
+[% TRY %][% INCLUDE 'report/_extras.html' %][% CATCH file %][% END %]
```
Comment (dracos) [inline templates/web/base/report/display.html]: Could this be called `_extras_top.html` or similar to signify it's included first thing?

File: templates/web/fixmystreet/footer.html [hunk 2]
```diff
@@ -37,5 +37,10 @@
 
 <!-- [% INCLUDE 'debug_footer.html' %] -->
     </div> <!-- .wrapper -->
+
+    [% IF c.cobrand.moniker == 'fixmystreet' %]
+        [% INCLUDE 'footer-research-js.html' %]
+    [% END %]
+
```
Comment (dracos) [inline templates/web/fixmystreet/footer.html]: This one is outstanding being moved to a separate template (`footer_extra.html` to match the header one?), as mentioned at https://github.com/mysociety/fixmystreet/commit/9ca588ae0#commitcomment-9217074 .

**Comment by dracos** [comment]
Yay, much nicer with separate templates :) One more needing doing, and a suggested rename of another.

**Comment by jacksonj04** [comment]
@dracos: Any further comments, or is this safe to merge and deploy? If it is, does FMS have any particular deployment process or schedule I should be aware of?

**Comment by dracos** [comment]
All squashed together, with one bugfix – it was `fixmystreet/report/display.html` that needed to be changed, not `base/report/display.html`. I have added `ms_srv_*` cookies to our Varnish configuration so that they are ignored for the purposes of backend caching.

**Pull request merged**